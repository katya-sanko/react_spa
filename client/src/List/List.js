import React from 'react';
/* add CARD component */

const List = ({ movies }) => (
    <ul>
        <h4>Results</h4>
        {
            movies && movies.map((movie, index) => <li key={index}>  <span>{movie.title}</span><br></br><mark>{movie.tagline}</mark>  </li>)
        }
    </ul>
);

export default List;