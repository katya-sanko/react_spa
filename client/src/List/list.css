.list-container {
    display: flex;
}

.card-wrapper {
    width: 33%;
}