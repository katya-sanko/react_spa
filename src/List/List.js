import React, { Component } from 'react';
import Card from '../Card/Card';
import './list.css';
import { connect } from 'react-redux';

class List extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        if (this.props.hasErrored) {
            return <span>Something went wrong...</span>
        }

        if (this.props.isLoading) {
            return <p>Loadingâ€¦</p>;
        }

        if (this.props.movies && !this.props.movies.length) {
            return <p>No films found</p>;
        }

        return (
            <div className="list-wrapper">
                <ul className="list-container">
                    {
                        this.props.movies && this.props.movies.map((movie, index) => <li className="card-wrapper" key={index}>  <Card movie={movie} />  </li>)
                    }
                </ul>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        movies: state.movies,
        hasErrored: state.moviesHasErrored,
        isLoading: state.moviesIsLoading
    };
};

export default connect(mapStateToProps)(List);