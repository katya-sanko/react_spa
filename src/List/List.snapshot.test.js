import React from 'react';
import List from './List';
import renderer from 'react-test-renderer';

const mockMovies = [];

it('renders correctly', () => {
    const tree = renderer
        .create(<List movies={mockMovies}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});