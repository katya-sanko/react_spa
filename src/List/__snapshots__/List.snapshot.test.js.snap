// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="list-wrapper"
>
  <ul
    className="list-container"
  >
    <li
      className="card-wrapper"
    >
        
      <div
        className="card"
      >
        <div
          className="card-poster"
        >
          <img
            className="card-posterImage"
            src="https://image.tmdb.org/t/p/w500/aMETsaNNcDc6g5ZatQtVbySnSaA.jpg"
          />
        </div>
        <div
          className="card-inlineInfo"
        >
          <span
            className="card-title"
          >
            Untitled Avengers
          </span>
          <span
            className="card-releaseYear"
          >
            2019
          </span>
        </div>
        <span
          className="card-genre"
        >
           
          Action
            
        </span>
        <span
          className="card-genre"
        >
           
          Adventure
            
        </span>
        <span
          className="card-genre"
        >
           
          Science Fiction
            
        </span>
      </div>
        
    </li>
    <li
      className="card-wrapper"
    >
        
      <div
        className="card"
      >
        <div
          className="card-poster"
        >
          <img
            className="card-posterImage"
            src="https://image.tmdb.org/t/p/w500/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg"
          />
        </div>
        <div
          className="card-inlineInfo"
        >
          <span
            className="card-title"
          >
            Avengers: Infinity War
          </span>
          <span
            className="card-releaseYear"
          >
            2018
          </span>
        </div>
        <span
          className="card-genre"
        >
           
          Adventure
            
        </span>
        <span
          className="card-genre"
        >
           
          Science Fiction
            
        </span>
        <span
          className="card-genre"
        >
           
          Fantasy
            
        </span>
        <span
          className="card-genre"
        >
           
          Action
            
        </span>
      </div>
        
    </li>
  </ul>
</div>
`;
